src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -304,19 +392,23 @@ public String getMonitoringEndpoint(Region region) {
     }
 
     /**
-     * Check if a metric should be used according to `aws_dimension_select` or `aws_dimension_select_regex`
+     * Check if a metric should be used according to `aws_dimension_select`, `aws_dimension_select_regex` and dynamic `aws_tag_select`
      */
-    private boolean useMetric(MetricRule rule, Metric metric) {
-      if (rule.awsDimensionSelect == null && rule.awsDimensionSelectRegex == null) {
+    private boolean useMetric(MetricRule rule, List<String> tagBasedResourceIds, Metric metric) {
+      if (rule.awsDimensionSelect == null && rule.awsDimensionSelectRegex == null && rule.awsTagSelect == null) {
         return true;
-      }
-      if (rule.awsDimensionSelect != null  && metricsIsInAwsDimensionSelect(rule, metric)) {
-        return true;
-      }
-      if (rule.awsDimensionSelectRegex != null  && metricIsInAwsDimensionSelectRegex(rule, metric)) {
+      } else {
```

Comment:
You can simplify here. The body of this else is enough.

Reply:
@brian-brazil Oh... indeed! I fixed it.